flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/overlays/FlinkDistributionOverlay.java

```diff
@@ -43,13 +43,14 @@
  * possible to bypass this overlay and rely on the normal installation method.
  *
  * <p>The following files are copied to the container:
- *  - flink/bin/
- *  - flink/conf/
- *  - flink/lib/
+ *  - bin/
+ *  - conf/
+ *  - lib/
+ *  - plugins/
  */
 public class FlinkDistributionOverlay extends AbstractContainerOverlay {
 
-	static final Path TARGET_ROOT = new Path("flink");
+	static final File TARGET_ROOT = new File(Path.CUR_DIR);
```

Comment:
Originally `FlinkDistributionOverlay` uses the `TARGET_ROOT` to set the `ENV_FLINK_HOME_DIR`.  But if the `Path TARGET_ROOT = new Path(Path.CUR_DIR)` the `TARGET_ROOT.toString()` is empty. 
This will lead `MESOS_TM_CMD`[1] to `/bin/mesos-taskmanager.sh` which can not launched the TM.


-----
[1] `public static final ConfigOption<String> MESOS_TM_CMD =
		key("mesos.resourcemanager.tasks.taskmanager-cmd")
		.defaultValue("$FLINK_HOME/bin/mesos-taskmanager.sh"); // internal`

Reply:
I see. This may considered to be a bug in the `Path` class. Since it is not obvious that `Path` cannot be used, can work around that by:

```
	private static final String TARGET_ROOT_STR = Path.CUR_DIR;

	static final Path TARGET_ROOT = new Path(TARGET_ROOT_STR);
```

```
container.getEnvironmentVariables().put(ENV_FLINK_HOME_DIR, TARGET_ROOT_STR);
```

Moreover, shouldn't there be a unit test in `FlinkDistributionOverlayTest` asserting that the environment variable is correctly set?

Reply:
OK. And I could add a test for that. It's a good suggestion.